src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -822,45 +831,38 @@ private async Task<SyntaxNode> ReplaceDiscardDeclarationsWithAssignmentsAsync(Sy
             {
                 // Get the current declaration statement.
                 var declStatement = memberDeclaration.GetCurrentNode(originalDeclStatement);
-
-                var documentUpdated = false;
+                Contract.ThrowIfNull(declStatement);
 
                 // Check if the variable declaration is unused after all the fixes, and hence can be removed.
                 if (await TryRemoveUnusedLocalAsync(declStatement, originalDeclStatement).ConfigureAwait(false))
                 {
-                    documentUpdated = true;
+                    await OnDocumentUpdatedAsync().ConfigureAwait(false);
                 }
                 else if (declStatement.HasAnnotation(s_newLocalDeclarationStatementAnnotation))
                 {
                     // Otherwise, move the declaration closer to the first reference if possible.
-                    if (await service.CanMoveDeclarationNearReferenceAsync(document, declStatement, cancellationToken).ConfigureAwait(false))
+                    if (await moveDeclarationService.CanMoveDeclarationNearReferenceAsync(document, declStatement, cancellationToken).ConfigureAwait(false))
                     {
-                        document = await service.MoveDeclarationNearReferenceAsync(document, declStatement, cancellationToken).ConfigureAwait(false);
-                        documentUpdated = true;
+                        document = await moveDeclarationService.MoveDeclarationNearReferenceAsync(document, declStatement, cancellationToken).ConfigureAwait(false);
+                        await OnDocumentUpdatedAsync().ConfigureAwait(false);
                     }
                 }
-
-                if (documentUpdated)
-                {
-                    await OnDocumentUpdatedAsync().ConfigureAwait(false);
-                }
             }
 
             return memberDeclaration;
 
             // Local functions.
             async Task OnDocumentUpdatedAsync()
             {
-                root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-                memberDeclaration = syntaxFacts.GetContainingMemberDeclaration(root, memberDeclaration.SpanStart);
+                root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                memberDeclaration = syntaxFacts.GetContainingMemberDeclaration(root, memberDeclaration.SpanStart) ?? root;
```

Comment:
this is the meat of the fix.  we were null-ref'ing later because there was no containing member, and we were nulling this out for the top level.